### ISSUE TITLE
Bug: [no-restricted-imports] Having the "allowTypeImports" option makes Eslint error on load

### ISSUE THREAD

**Issue by RamonBalthazar** [typescript-eslint/typescript-eslint#5514]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=JYWwDg9gTgLgBDAnmApnA3nAwhcEB2K+MAClBGAM5wC+cAZuSHAOQAClMwMAxgBYpKAeigoAhjxgsA3AChQkWHABK4yQyatREqXPl4lmTogA2KACa0NuVgD8ArsCHGzlGbNk8CnOAGUAbgDmcAC8cC4WABQslEEsADQYsnBwAsCBfDAAXKwATAAMYAAeCckMwCYmOSwAJJR8wPQwAIy5pTQAlHpe+D4AMhCBEDmqOgB0AGJYADw4eITEZBSU00ioEPR+QQB826FwkR2he+hlojD2UPhw+PaVsjR6KEWK8OYo9GJ38ANDckA&eslintrc=N4KABGBECmB2BukBcZQQpARgJwPYHcBnabZMAF2wFdoAacdGQgNjMpvvSllwBNo21OgwwAraIXKCaDAL6coABwCG2YqRSQAAuQCeiiQGNsAS0XkAtBIA2J2OQD0KtSUgLIi61QDmdwmQBtESgdfSNTcytCW3s3YMhCEwBbT2gLZMVcbEtCLKkGAF13aAAPcjhefxQgrkhsaGVDS2VFRTja+sbm1odxSXbGRXrychNXQvdqawkyNEYARypccqqwGq4ofFVYAdrE2G9p3fQ5jahleFwTXgBRQkMWgRR2YTOMZWtrAgAVaBTrZTlAAyJnK2A+qwAZhDoMEILJgkV4jwLPVJKYmtBeOkUnlVutaiQ8KQFFxThsPICABb4uEbclvbjKJJPEKSUGGKkSBydJrHRmQFmEQjKbysyAASVx2TAoLAkLwSTAAHIAH5UEwOSS6aaEZWy2CSBq8AB0-LekA+X3w3zCUsy2VWLzpXARjIKdLd6A98PcxBGdm8q3JkAyeR5Elw1ngrhQDO4fFZ8YwpXKhpMuENgRdjBNon8pItecIJXNtRN5ALOYwFZLkBzPo2XvhchAsiAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTokCskANODUipkAI3YBtATSgATBHymC58AHTI8WeiIwLptTkXQAPNooC6-aZHqpUsAO4ApMpTyYArukuCSAa2SIADKiAMIAFujQvuxunt76JACysDLuGACSRBqYsIgxHl6K1rYOAMoAnlkRBNAAIugAZvRpeOlIsJh4LmCxhVYkbsjQbK4F8VANHdDoIcQkyAPoWSH080RMmQBiaOgAcvTwnPlxRUSwmzaoeGE57kxhKyScmaX26tBhx32C8Clp6OxIAZjGxxpBfqkMAAlTiwVDuAjEQFnGQAsGYWGoABu6GcxGSkIBoxOVgWcPoGhkBP+3V6YLOAFF4OovmCAFYkIyAjH0YYAWg5XLBIlW6AAqphUICVAB6EiYaC6QSIClhbrUPSQAB+ACoJIorHqDWAzIoAL4CM3eSDIIjQeGo-X9BWQASms1AA

### Repro Code

```typescript
import type { ComponentProps } from '@stitches/react';
```

### ESLint Config

```javascript
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "jest": true
  },
  "parser": "@typescript-eslint/parser",
  "plugins": [
    "@typescript-eslint",
    "simple-import-sort"
  ],
  "extends": [
    "react-app",
    "react-app/jest",
    "prettier"
  ],
  "rules": {
    "no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "@stitches/react",
            "message": "Import it from '~ui/styles' instead.",
            "allowTypeImports": true
          }
        ]
      }
    ]
  },
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".jsx",
          ".ts",
          ".tsx"
        ]
      }
    }
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": "./src",
    "paths": {
      "~*": ["*"]
    }
  },
  "include": ["src"]
}
```

### Expected Result

I expected eslint to load and run.

### Actual Result

Eslint doesn't load and shows this error:

```
Oops! Something went wrong! :(

ESLint: 8.22.0

Error: .eslintrc.js:
	Configuration for rule "no-restricted-imports" is invalid:
	Value {"paths":[{"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true}]} should be string.
	Value {"paths":[{"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true}]} should NOT have additional properties.
	Value {"paths":[{"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true}]} should match some schema in anyOf.
	Value {"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true} should be string.
	Value {"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true} should NOT have additional properties.
	Value {"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true} should match some schema in anyOf.
	Value [{"paths":[{"name":"@stitches/react","message":"Import it from '~ui/styles' instead.","allowTypeImports":true}]}] should match some schema in anyOf.
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.33.1` |
| `@typescript-eslint/parser`            | `5.33.1` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.22.0` |
| `node`                                 | `18.6.0` |

**Comment by bradzacher** [comment]
https://typescript-eslint.io/docs/linting/troubleshooting/#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

The rule name is `@typescript-eslint/no-restricted-imports`. Note the prefix - it's required - otherwise you're using the base rule which doesn't support our new options (hence the error).
